exchange/exchange.go

```diff
@@ -344,7 +351,12 @@ func (e *exchange) getAllBids(ctx context.Context, cleanRequests map[openrtb_ext
 	// Wait for the bidders to do their thing
 	for i := 0; i < len(cleanRequests); i++ {
 		brw := <-chBids
-		adapterBids[brw.bidder] = brw.adapterBids
+
+		//if bidder returned no bids back - remove bidder from further processing
+		if brw.adapterBids != nil && len(brw.adapterBids.bids) != 0 {
```

Comment:
What are the consequences of removing the adapter bids entirely (do the metrics still record properly?) versus setting the bids to an empty slice?

Reply:
We don't want to carry bidder data if there are no bids back. For loops will skip empty array iterating and nothing bad will happen. This is just memory and processor time optimization. Hope this makes sense.